product-service/handlers/getProducts.test.ts

```diff
@@ -0,0 +1,48 @@
+import { describe, test, expect, jest } from '@jest/globals';
+import { getProducts } from './getProducts';
+import { ProductService } from '../services/product-service';
+
+jest.mock<typeof import('../services/product-service')>('../services/product-service');
+const mockMethod = jest.fn<() => any>();
+
+const getProductByIdMock = jest
+  .spyOn(ProductService.prototype, 'getProducts')
+  .mockImplementation(() => {
+    return mockMethod();
+  }); 
+
+describe('getProductById', () => {     
+    test('Should return empty array when there is no data', async () => {
+        mockMethod.mockReturnValue(null);
+        const data = await getProducts();
+        expect(data).toEqual(
+            {
+                "body": "[]",
+                "headers": {
+                    "Access-Control-Allow-Headers": "*",
+                    "Access-Control-Allow-Methods": "*",
+                    "Access-Control-Allow-Origin": "*",
+                },
+                "statusCode": 200,
+            }
+        );
+        expect(getProductByIdMock).toHaveBeenCalled();
+    })
+
+    test('Should return correct data for productid from getProdutById', async () => {
```

Comment:
test description should be something like  'should return data when response is not empty'

Reply:
Updated.